packages/nx/src/utils/create-package-json.ts

```diff
@@ -15,11 +30,66 @@ export function createPackageJson(
   projectName: string,
   graph: ProjectGraph,
   options: {
+    target?: string;
     root?: string;
     isProduction?: boolean;
-  } = {}
+  } = {},
+  additionalDependencies?: string[]
 ): PackageJson {
-  const npmDeps = findAllNpmDeps(projectName, graph);
+  let selfInputs: string[] = [];
+  let dependencyInputs: string[] = [];
+
+  if (options.target) {
+    const projectNode = graph.nodes[projectName];
+
+    const nxJson = readNxJson();
+
+    const namedInputs = {
+      ...DEFAULT_NAMED_INPUTS,
+      ...nxJson.namedInputs,
+      ...projectNode.data.namedInputs,
+    };
+    const targetData = projectNode.data.targets[options.target];
```

Comment:
I would move this entire logic to hasher and have a simple function to return `selfInputs` and `dependencyInputs` in an object.

That would encapsulate the entire logic away.